ovs/extensions/services/service.py

```diff
@@ -60,3 +80,54 @@ def __getattr__(cls, item):
             return getattr(ServiceManager.ImplementationClass, item)
 
     __metaclass__ = MetaClass
+
+    @staticmethod
+    def reload():
+        ServiceManager.ImplementationClass = None
+
+    @staticmethod
+    def has_fleet_client():
+        return HAS_FLEET_CLIENT
+
+    @staticmethod
+    def setup_fleet():
+        if HAS_FLEET_CLIENT is False:
+            logger.info('Cannot use fleet because the client is not installed')
+            return False
+        if ServiceManager.has_fleet():
+            if ServiceManager._is_fleet_running_and_usable():
+                logger.info('Fleet service is running')
+                ServiceManager.reload()
+                return True
+            else:
+                check_output('systemctl start fleet', shell=True)
+                start = time.time()
+                while time.time() - start < 15:
+                    if ServiceManager._is_fleet_running_and_usable():
+                        logger.info('Fleet service is running and usable')
+                        ServiceManager.reload()
+                        return True
+                    time.sleep(1)
+                raise RuntimeError('Can not use fleet to manage services.')
+
+    @staticmethod
+    def has_fleet():
+        try:
+            has_fleetctl_bin = 'fleetctl' in check_output('which fleetctl', shell=True)
+            has_fleetd_bin = 'fleetd' in check_output('which fleetd', shell=True)
+            return has_fleetctl_bin and has_fleetd_bin
+        except CalledProcessError as cpe:
+            logger.warning('Could not determine if fleet can be used. {0}'.format(cpe))
+            return False
+
+    @staticmethod
+    def _is_fleet_running_and_usable():
+        try:
+            is_fleetd_running = 'active' in check_output('systemctl is-active fleet || true', shell=True).strip()
```

Comment:
I dont see the added value to first check if fleet is active
If you know the binaries are both present, you can try to execute the list-machines command right away --> if this succeeds return True else False